src/cudamatrix/cu-matrix.cc

```diff
@@ -2528,8 +2528,45 @@ void CuMatrixBase<Real>::AddElements(Real alpha,
 }
 
 template<typename Real>
+void CuMatrixBase<Real>::AddElements(Real alpha, const CuArray<Int32Pair> &indexes,
+				     const Real *input) {
+  KALDI_ASSERT(input != NULL);
+  // Checks the dimension.
+  MatrixIndexT num_rows = this->num_rows_, num_cols = this->num_cols_;
+  std::vector<Int32Pair> cpu_indexes;
+  indexes.CopyToVec(&cpu_indexes);
+  for (int32 i = 0; i < indexes.Dim(); ++i) {
+    KALDI_ASSERT(cpu_indexes[i].first < num_rows && cpu_indexes[i].first >= 0 &&
+                 cpu_indexes[i].second < num_cols && cpu_indexes[i].second >= 0);
+  }
+
+#if HAVE_CUDA == 1
+  if (CuDevice::Instantiate().Enabled()) {
+    void *addr = CuDevice::Instantiate().Malloc(indexes.Dim() * sizeof(Real));
```

Comment:
'addr' is not a great name, something like cu_array might be more suitable.  However, in situations like these I prefer you to do something like
  Vector<Real> tmp_vec(indexes.Dim(), kUndefined);
  and then use tmp_vec.Data().   This automatically takes care of error handling and deallocation.